routingpy/routers/valhalla.py

```diff
@@ -282,6 +282,7 @@ def parse_direction_json(response, units):
             factor = 0.621371 if units == "mi" else 1
             distance += int(leg["summary"]["length"] * 1000 * factor)
 
+        geometry = [coord[::-1] for coord in geometry]
```

Comment:
hm, lat/lon is actually the wrong order. `Direction` object exposes lon,lat tuples by definition, which plays much nicer with virtually all geospatial formats & libs except the oddballs leaflet, google maps api and very few more.

Reply:
how can this pass the tests though?!

Reply:
I wonder if it wouldn't be a good idea to review how the tests are made. I'd like to see a snapshot system of API responses that can be easily updated. And we could also add live tests in the pipeline if we manage not to have all the API keys and not to make too many requests. What do you think?

Reply:
> My Google routes are displayed, but not Valhalla's.

That'd be surprising.. I didn't work intensely with routingpy in a while, but we're using it extensively in other projects with Valhalla specifically and I'd hope we'd have caught such an obvious error. I'll have a try later myself. In any case, if there's a problem, it's the polyline decoding and it should be fixed there (utils.py).

> And we could also add live tests in the pipeline if we manage to have all the API keys and not to make too many requests

Yeah, that was heavily worked on a while ago by @chrstnbwnkl : https://github.com/gis-ops/routingpy/pull/85. However, there was a problem with ORS IIRC. And then priorities shifted massively and it didn't make it into master yet.

Reply:
@nilsnolde Don't bother looking. After a second look, I think the problem is on my side 😅

Reply:
In the end it was Google's that were reversed